src/main/java/reactor/ipc/netty/ReactorNetty.java

```diff
@@ -112,4 +303,27 @@ protected void channelIdle(ChannelHandlerContext ctx,
 	}
 
 	static final Object TERMINATED = new TerminatedHandlerEvent();
+	static final Logger log        = Loggers.getLogger(ReactorNetty.class);
+
+	/**
+	 * A handler that can be used to extract {@link ByteBuf} out of {@link ByteBufHolder},
+	 * optionally also outputting additional messages
+	 *
+	 * @author Stephane Maldini
+	 * @author Simon Baslé
+	 */
+	@ChannelHandler.Sharable
+	static final class ExtractorHandler extends ChannelInboundHandlerAdapter {
+
+		final BiConsumer<? super ChannelHandlerContext, Object> extractor;
+
+		ExtractorHandler(BiConsumer<? super ChannelHandlerContext, Object> extractor) {
+			this.extractor = Objects.requireNonNull(extractor, "extractor");
+		}
+
+		@Override
+		public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+			extractor.accept(ctx, msg);
```

Comment:
this is quite opinionated and already the http extractor has issues with the right type of message (not only last frame but start frame). Also the Consumer can condition the emission of some frame if required (empty for instance).